runelite-client/src/main/java/net/runelite/client/plugins/blowpipe/BlowpipePlugin.java

```diff
@@ -0,0 +1,305 @@
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameObjectSpawned;
+import net.runelite.api.Projectile;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.ProjectileMoved;
+import net.runelite.api.events.AnimationChanged;
+import java.util.regex.Matcher;
+import java.text.ParseException;
+import java.text.NumberFormat;
+import java.util.Locale;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.api.Item;
+import net.runelite.api.InventoryID;
+import net.runelite.api.kit.KitType;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.ui.overlay.Overlay;
+import java.util.regex.Pattern;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.OverlayManager;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import org.slf4j.Logger;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.Plugin;
+
+@PluginDescriptor(
+        name = "Blowpipe Helper",
+        description = "Show various helpful utitiles during the Vorkath fight",
+        tags = { "bosses", "combat", "minigame", "overlay", "prayer", "pve", "pvm" },
+        type = PluginType.PVM
+        )
+public class BlowpipePlugin extends Plugin
+{
+    private static final Logger log = LoggerFactory.getLogger(BlowpipePlugin.class);
+    @Inject
+    private Client client;
+    @Inject
+    private OverlayManager overlayManager;
+    @Inject
+    private ItemManager itemManager;
+    @Inject
+    private BlowpipeInfoOverlay blowpipeInfoOverlay;
+    private boolean show;
+    private BlowpipeProjectile currentDartType;
+    private int currentDartEstimate;
+    private int currentScaleEstimate;
+    private int dartsInInventory;
+    private static Pattern BLOWPIPE_AMMO = Pattern.compile("Darts: <col=007f00>(.+)</col>. Scales: <col=007f00>(.+)</col>");
+
+
+    public BlowpipePlugin() {
+        this.currentDartType = null;
+        this.currentDartEstimate = -1;
+        this.currentScaleEstimate = -1;
+        this.dartsInInventory = 0;
+    }
+
+    @Override
+    protected void startUp() throws Exception {
+        overlayManager.add(blowpipeInfoOverlay);
+        currentDartEstimate = -1;
+        currentScaleEstimate = -1;
+        currentDartType = null;
+    }
+
+    @Override
+    protected void shutDown() throws Exception {
+        overlayManager.remove(blowpipeInfoOverlay);
+        currentDartEstimate = -1;
+        currentScaleEstimate = -1;
+        currentDartType = null;
+    }
+
+    @Subscribe
+    public void onGameTick(GameTick event)
+    {
+        show = false;
+        dartsInInventory = 0;
+        int weapon = client.getLocalPlayer().getPlayerComposition().getEquipmentId(KitType.WEAPON);
+        Item[] items = client.getItemContainer(InventoryID.INVENTORY).getItems();
```

Comment:
Why check inventory/worn on game tick? There's events for item containers changing